File: go/algorithms/comb.go
package algorithms

// CombSort sorts in-place using shrinking gap.
func CombSort(arr []int) {
    n := len(arr)
    gap := n
    swapped := true
    for gap > 1 || swapped {
        gap = int(float64(gap) / 1.3)
        if gap < 1 {
            gap = 1
        }
        swapped = false
        for i := 0; i+gap < n; i++ {
            if arr[i] > arr[i+gap] {
                arr[i], arr[i+gap] = arr[i+gap], arr[i]
                swapped = true
            }
        }
    }
}

File: rust/src/comb.rs
pub fn comb_sort(arr: &mut [i32]) {
    let mut gap = arr.len();
    let mut swapped = true;
    while gap > 1 || swapped {
        gap = (gap as f64 / 1.3).floor() as usize;
        if gap < 1 {
            gap = 1;
        }
        swapped = false;
        for i in 0..arr.len().saturating_sub(gap) {
            if arr[i] > arr[i + gap] {
                arr.swap(i, i + gap);
                swapped = true;
            }
        }
    }
}
